mx.truffle/mx_truffle.py

```diff
@@ -93,3 +94,58 @@ def _truffle_gate_runner(args, tasks):
     'sldebug' : [sldebug, '[SL args|@VM options]'],
     'testdownstream' : [testdownstream, ''],
 })
+
+"""
+Merges META-INF/truffle/language and META-INF/truffle/instrumentation files.
+This code is tightly coupled with the file format generated by
+LanguageRegistrationProcessor and InstrumentRegistrationProcessor.
+"""
+class TruffleArchiveParticipant:
+    PROPERTY_RE = re.compile(r'(language\d+|instrumentation\d+)(\..+)')
+
+    def _truffle_metainf_file(self, arcname):
+        if arcname == 'META-INF/truffle/language':
+            return 'language'
+        if arcname == 'META-INF/truffle/instrumentation':
+            return 'instrumentation'
+        return None
+
+    def __opened__(self, arc, srcArc, services):
+        self.settings = {}
+        self.arc = arc
+
+    def __add__(self, arcname, contents):
+        metainfFile = self._truffle_metainf_file(arcname)
+        if metainfFile:
+            properties = self.settings.setdefault(metainfFile, {})
+            for line in contents.strip().split(os.linesep):
+                if not line.startswith('#'):
+                    m = TruffleArchiveParticipant.PROPERTY_RE.match(line)
+                    assert m, line
+                    enum = m.group(1)
+                    prop = m.group(2)
+                    properties.setdefault(enum, []).append(prop)
+            return True
+        return False
+
+    def __addsrc__(self, arcname, contents):
+        return False
+
+    def __closing__(self):
+        for metainfFile, properties in self.settings.iteritems():
+            arcname = 'META-INF/truffle/' + metainfFile
+            lines = []
+            counter = 1
+            for enum in sorted(properties.viewkeys()):
+                assert enum.startswith(metainfFile)
+                newEnum = metainfFile + str(counter)
+                counter += 1
+                for prop in properties[enum]:
+                    lines.append(newEnum + prop)
+
+            content = '\n'.join(lines)
```

Comment:
`Properties.store0` uses the platform dependent line separator. On the other hand, when loading a properties file, Java handles both. In any case, I will implement your request Christian.